prow/cmd/phaino/local.go

```diff
@@ -246,23 +254,66 @@ func convertToLocal(ctx context.Context, log *logrus.Entry, pj prowapi.ProwJob,
 		log.Warn("Ignoring resource requirements")
 	}
 
-	for _, mount := range container.VolumeMounts {
-		vol := volume(*pj.Spec.PodSpec, mount.Name)
-		if vol == nil {
-			return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
+	if opt.mountJobVolumes {
+		for _, mount := range container.VolumeMounts {
+			vol := volume(*pj.Spec.PodSpec, mount.Name)
+			if vol == nil {
+				return nil, fmt.Errorf("mount %q missing associated volume", mount.Name)
+			}
+			if vol.EmptyDir != nil {
+				localArgs = append(localArgs, "-v", mount.MountPath)
+			} else {
+				local, err := readMount(ctx, mount)
+				if err != nil {
+					return nil, fmt.Errorf("bad mount %q: %v", mount.Name, err)
+				}
+				arg := local + ":" + mount.MountPath
+				if mount.ReadOnly {
+					arg += ":ro"
+				}
+				localArgs = append(localArgs, "-v", arg)
+			}
 		}
-		if vol.EmptyDir != nil {
-			localArgs = append(localArgs, "-v", mount.MountPath)
+	}
+
+	// Setup gcloud credentials
+	if opt.useLocalGcloudCredentials {
```

Comment:
In the case the prow job already defined a volume mount + envvar `GOOGLE_APPLICATION_CREDENTIALS`, the desired behavior would be phaino is smart enough to mount for me and not ask at previous mount step, this also applies to KUBECONFIG. WDYT?

Reply:
That is a good idea! 

I can change the line to `if opt.useLocalGcloudCredentials || jobSpec.GetEnv("GOOGLE_APPLICATION_CREDENTIALS") != ""`, since if the env var is defined, it assumes there must be a volume mount. Does it make sense to you?

Reply:
I would want the other way around: don't ask me to mount GCP SA key at previous step because I already said use default credentials.

Reply:
Done.